scripts/build-packages-dist.mts

```diff
@@ -0,0 +1,155 @@
+  const targets = exec(queryPackagesCmd, true)
+    .split(/\r?\n/)
+    .map((target) => target + '_archive');
+  const packageNames = getPackageNamesOfTargets(targets);
+  const bazelBinPath = exec(`${bazelCmd} info bazel-bin`, true);
+  const getBazelOutputPath = (pkgName: string) =>
+    join(bazelBinPath, 'packages', pkgName, 'npm_package');
+  const getDistPath = (pkgName: string) => join(distPath, pkgName);
+
+  // Build with "--config=release" or `--config=snapshot` so that Bazel
+  // runs the workspace stamping script. The stamping script ensures that the
+  // version placeholder is populated in the release output.
+  const stampConfigArg = `--config=${isSnapshotBuild ? 'snapshot' : 'release'}`;
+
+  // Walk through each release package and clear previous "npm_package" outputs. This is
+  // a workaround for: https://github.com/bazelbuild/rules_nodejs/issues/1219. We need to
+  // do this to ensure that the version placeholders are properly populated.
+  packageNames.forEach((pkgName) => {
+    const outputPath = getBazelOutputPath(pkgName);
+    if (sh.test('-d', outputPath)) {
+      sh.chmod('-R', 'u+w', outputPath);
+      sh.rm('-rf', outputPath);
+    }
+  });
+
+  exec(`${bazelCmd} build ${stampConfigArg} ${targets.join(' ')}`);
+
+  // Delete the distribution directory so that the output is guaranteed to be clean. Re-create
+  // the empty directory so that we can copy the release packages into it later.
+  rmSync(distPath, { force: true, recursive: true, maxRetries: 3 });
+  mkdirSync(distPath, { recursive: true });
+
+  // Copy the package output into the specified distribution folder.
+  packageNames.forEach((pkgName) => {
+    const outputPath = getBazelOutputPath(pkgName);
+    const targetFolder = getDistPath(pkgName);
+    console.log(`> Copying package output to "${targetFolder}"`);
+
+    // Ensure package scope directory exists prior to copying
+    mkdirSync(dirname(targetFolder), { recursive: true });
+
+    // Copy package contents to target directory
+    sh.cp('-R', outputPath, targetFolder);
+    sh.chmod('-R', 'u+w', targetFolder);
+
+    // Copy archive of package to target directory
+    const archiveTargetPath = join(distPath, `${pkgName.replace('/', '_')}.tgz`);
+    copyFileSync(join(dirname(outputPath), 'npm_package_archive.tgz'), archiveTargetPath);
+    chmodSync(archiveTargetPath, '0755');
+  });
+
+  return packageNames.map((pkg) => {
+    return {
+      // Package names on disk do not have the @ scope prefix and use underscores instead of dashes
+      name: `@${pkg.replace(/_/g, '-')}`,
+      outputPath: getDistPath(pkg),
+    };
+  });
+}
+
+/**
+ * Gets the package names of the specified Bazel targets.
+ * e.g. //packages/angular/cli:npm_package = angular/cli
+ */
+function getPackageNamesOfTargets(targets: string[]): string[] {
+  return targets.map((targetName) => {
+    const matches = targetName.match(/\/\/packages\/(.*):npm_package/);
```

Comment:
I believe this just works currently because `npm_package_archive` matches too. May want to clarify the comment above, or be explicit `:npm_package(_archive)?$`